### PR TITLE
Support for uint

### PR DESCRIPTION
**Pull request by roshats**
I've noticed that support for uint was missing, so I've added it in this pull request. Let me know if any changes are needed.

### CODE REVIEW

File: reader_test.go [hunk 1]
```diff
@@ -101,9 +101,9 @@ Bob,bob@example.com,25
 
 func TestComplexTypes(t *testing.T) {
 	// CSV data with various types
-	csvData := `name,created_at,active,score,count,rate,tags
-John,2023-01-02T15:04:05Z,true,98.6,42,3.14,test;debug
-Jane,2023-06-15T09:30:00Z,false,75.2,100,2.718,prod;live`
+	csvData := `name,created_at,active,score,count,rate,tags,rank
+John,2023-01-02T15:04:05Z,true,98.6,42,3.14,test;debug,1
+Jane,2023-06-15T09:30:00Z,false,75.2,100,2.718,prod;live,42`
```
Comment (ccoVeille) [inline reader_test.go:106]: Could you add a test about sending a negative number via a CSV into a uint field ?

I feel like it could be useful to validate how the library behaves in such use case.

Also, it would help to avoid regression

**Review by ccoVeille — APPROVED** [review]
LGTM 👍

It's a legitimate use case.

One minor remark anyway.

File: reader_test.go (hunk 1 above)
Comment (roshats) [reply reader_test.go:106]: Thanks for the comment. I've added a test for invalid values.

File: reader_test.go [hunk 2]
```diff
@@ -166,3 +168,48 @@ func TestCustomUnmarshaler(t *testing.T) {
...
+		})
+
+		if actualValue == nil {
+			t.Errorf("Expected a panic for input '%s', but it was parsed without error", input)
+			return
+		}
+
+		actualError, ok := actualValue.(error)
+		if !ok {
+			t.Errorf("Expected a panic for input '%s', but received a non-error value: %v", input, actualValue)
```
Comment (ccoVeille) [inline reader_test.go]: Maybe this

```suggestion
			t.Errorf("Expected a panic for input '%s', but received a non-error value: %v (%[2]T)", input, actualValue)
```

File: reader_test.go [hunk 3]
```diff
@@ -166,3 +168,48 @@ func TestCustomUnmarshaler(t *testing.T) {
...
+			return
+		}
+
+		actualMessage := actualError.Error()
+		if expectedMessage != actualMessage {
+			t.Errorf("Expected a panic for input '%s' with error message '%s', but got '%s'", input, expectedMessage, actualMessage)
```
Comment (ccoVeille) [inline reader_test.go]: Here and everywhere, you could use %q


```suggestion
			t.Errorf("Expected a panic for input %q with error message %q, but got %q", input, expectedMessage, actualMessage)
```

File: reader_test.go [hunk 4]
```diff
@@ -166,3 +168,48 @@ func TestCustomUnmarshaler(t *testing.T) {
...
+		defer func() {
+			value = recover()
+		}()
+
+		f()
+		return
+	}
+
+	test := func(t *testing.T, input string, expectedMessage string) {
+		reader := strings.NewReader(input)
```
Comment (ccoVeille) [inline reader_test.go]:
```suggestion
	test := func(t *testing.T, input string, expectedMessage string) {
		t.Helper()

		reader := strings.NewReader(input)
```

File: reader_test.go [hunk 5]
```diff
@@ -166,3 +168,48 @@ func TestCustomUnmarshaler(t *testing.T) {
...
+		}
+
+		actualMessage := actualError.Error()
+		if expectedMessage != actualMessage {
+			t.Errorf("Expected a panic for input '%s' with error message '%s', but got '%s'", input, expectedMessage, actualMessage)
+		}
+	}
+
+	test(t, "score\nnot-float", `error setting field Score: strconv.ParseFloat: parsing "not-float": invalid syntax`)
+	test(t, "rank\n-1", `error setting field Rank: strconv.ParseUint: parsing "-1": invalid syntax`)
```
Comment (ccoVeille) [inline reader_test.go]: Here the test are too dependent from the implementation. You can see the error message is about strconv failing to parse.

I'm doing this comment here, but I mean it's interesting to change the implementation, and so the tests

File: shared_test.go [hunk 6]
```diff
@@ -21,6 +21,7 @@ type ComplexRecord struct {
 	Count     int       `csv:"count"`
 	Rate      float32   `csv:"rate"`
 	Tags      string    `csv:"tags"`
+	Rank      uint      `csv:"rank"`
```
Comment (ccoVeille) [inline shared_test.go]: Please use a uint8 so you could also use 1000 as an invalid value, and simulate an issue an int64 might have with a huge number

File: reader_test.go (hunk 5 above)
Comment (ccoVeille) [reply reader_test.go]: Never mind ... The whole lib is like that...
It's out of the scope of this PR

File: shared_test.go (hunk 6 above)
Comment (roshats) [reply shared_test.go]: I've added overflow check for both uint and int.

**Review by ccoVeille — APPROVED** [review]
